### ISSUE TITLE
Allow chromecast control over playlist - Almost working feature.

### ISSUE THREAD

**Issue by sergioavazquez** [mediaelement/mediaelement-plugins#81]
I downloaded the latest version of mediaelement-plugins, the npm version doesn't have the playlist folder on dist... I'm assuming it's because it's not updated it yet.

I found out a way to change channels using the playlist while casting to chromecast with a minor tweak. 
On mediaelement-plugins/src/playlist/playlist.js line 171, the `player.load()` causes problems with chromecast's session.

I commented that line and tested it on PC and android with and without chromecast and it works great. The only caveat is that all the streams I'm using have the same mime type for HLS. I haven't tested if it works when using different stream formats.

Is this method call necessary? If not, can it be removed so it can work while casting? 
If it can't be removed, can you think of another easy implementation to allow switching channels while casting?

The only other detail I found that could be improved for this feature is that data-cast-... arguments are not being read from the list of sources, so data-cast-title is always the one defined on the video element, or none if you did not set one up.

I think we might be able to add an interesting new feature with a few minor tweaks.. Thoughts?

**Comment by rafa8626** [comment]
If you can provide me a URL so I can see your issue I'd appreciate it. The method is necessary otherwise it won't load the media (at least on my tests wit Mac OSX, without it doesn't work)

**Comment by rafa8626** [comment]
BTW I have reached Chromecast team because I have issues switching from one type to another. I didn't see any issues with the `player.load` when I set all my sources to video/mp4

**Comment by sergioavazquez** [comment]
I'm using ubuntu and chrome. I've been meaning to get this done for a while now, so I had already tested changing media just by changing the source with the `player.setSrc()` method. Always HLS. It worked great so for me this was a done deal. 

When I tried it again yesterday with the playlist plugin, chromecast threw a few session errors and it took quite a while to start casting on occasions. Similar to the behavior we encountered before we found how to handle cast sessions. I checked and saw you were using the load method. When removed, casting media changes work flawlessly for me, whether I'm casting or not. Better yet, changing content is a lot faster... 

I wouldn't be surprised if this is a Mac specific thing... or if it's related to the mime type..  I'm going to keep testing. I'll give you access to it as soon as I can.

**Comment by rafa8626** [comment]
@sergioavazquez Just FYI I won't be maintaining MediaElement anymore. If I have any info about this I'll let you know. Thanks for all your help with Chromecast

**Comment by sergioavazquez** [comment]
@rafa8626 Thank you for you help and good luck with whatever you do next!

**Comment by rafa8626** [comment]
@sergioavazquez Just for curiosity, did you ever find a solution for this? Google sent me an email saying that the errors were related to the stream itself and how Chromecast processed it, but that's it.

**Comment by sergioavazquez** [comment]
@rafa8626 I decided on not using the playlist plugin and develop my own list like component and apply the method described above to change the content source. It works everywhere I tested. But I'm not using Air Play or any apple related product to "cast". Changing contents on iOS works great though!
Still couldn't hack updating content while casting if the mime type changes from HLS to DASH. It simply doesn't start. It could be the stream though, I haven't tested this thoroughly yet.
